### PR TITLE
JSAB2-15 Academy

### PR DESCRIPTION
**Pull request by HONGTAO-T**
when you click on your academy, you will see a popup. Click on the create troop button, you will receive a notification on your phone and your level 1 troop will increase by 1.

### CODE REVIEW

**Review by kyya — APPROVED** [review]

**Review by HanqingZ — APPROVED** [review]

File: backend/routers/troops.js [hunk 1]
```diff
@@ -18,8 +21,13 @@ async function createTroop(req, res, next) {
   const { _id: owner } = req.user;
   const level = Number.parseInt(req.query.level, 10) || 1;
   try {
+    const cost = parseInt(foxRule.constructionCost, 10);
+    const purchase = await ResourceModel.purchaseItem(owner, cost);
```
Comment (evasimonyi) [inline backend/routers/troops.js]:
```suggestion
    const hasEnoughMoney = await ResourceModel.purchaseItem(owner, cost);
```

purchase is not describing what you are checking

File: frontend/src/Troops/actionCreator.js [hunk 2]
```diff
@@ -25,3 +28,26 @@ export function fetchTroops(token) {
       .catch((error) => dispatch({ type: FETCH_TROOPS_FAILURE, payload: error }));
   };
 }
+
+export function createTroop(level, token) {
+  return (dispatch) => {
+    dispatch({ type: CREATE_TROOP_REQUEST });
+    return fetch(`${URL}?level=${level}`, {
```
Comment (evasimonyi) [inline frontend/src/Troops/actionCreator.js]:
```suggestion
    return fetch(`${URL}/?level=${level}`, {
```

Comment (evasimonyi) [reply frontend/src/Troops/actionCreator.js]: did it work for you this way without the `/`?

File: frontend/src/Troops/index.jsx [hunk 3]
```diff
@@ -22,36 +22,36 @@ function Troops() {
...
 
-  if (isLoading || !troop) {
+  if (isLoading || Object.keys(listOfTroops.length === 0)) {
     return <ActivityIndicator size="large" color={Colors.tealColor} />;
   }
 
   const {
     attack, defence, hp: sustenance, countByLevel,
-  } = troop;
+  } = listOfTroops;
```
Comment (evasimonyi) [inline frontend/src/Troops/index.jsx]: it's rather details of troops and not an actual list, isn't it? as I see it's not the troops but the information about them so the name is confusing

**Review by evasimonyi — CHANGES_REQUESTED** [review]

File: frontend/src/Troops/index.jsx [hunk 4]
```diff
@@ -20,38 +20,38 @@ function Troops() {
...
-
-  const [troop] = listOfTroops;
+  const { infoOfTroops, isLoading, error } = useSelector((state) => state.troops);
 
   if (error) {
     return <ErrorPopup message={`Oops, ${error.message}`} />;
   }
 
-  if (isLoading || !troop) {
+  if (isLoading || Object.keys(infoOfTroops.length === 0)) {
```
Comment (kyya) [inline frontend/src/Troops/index.jsx]:
```suggestion
  if (isLoading || Object.keys(infoOfTroops).length === 0) {
```

**Review by kyya — CHANGES_REQUESTED** [review]

**Review by HanqingZ — APPROVED** [review]

**Review by GuJialu — APPROVED** [review]

**Review by kyya — APPROVED** [review]

File: backend/routers/troops.js [hunk 5]
```diff
@@ -18,8 +21,13 @@ async function createTroop(req, res, next) {
   const { _id: owner } = req.user;
   const level = Number.parseInt(req.query.level, 10) || 1;
   try {
+    const cost = parseInt(foxRule.constructionCost, 10);
+    const hasEnoughMoney = await ResourceModel.purchaseItem(owner, cost);
```
Comment (evasimonyi) [inline backend/routers/troops.js]:
```suggestion
    const hasEnoughMoney = await ResourceModel.purchaseItem(owner, cost);
```

don't have `purchaseItem` method anymore, please rebase

**Review by evasimonyi — CHANGES_REQUESTED** [review]

**Review by HanqingZ — APPROVED** [review]

**Review by kyya — APPROVED** [review]

**Review by evasimonyi — APPROVED** [review]

**Pull request merged**